public/js/angular/filters/parsable_date.js

```diff
@@ -1,21 +1,25 @@
 angular.module('parseDate', [])
 .filter('parsableDate', function() {
     return function(input) {
-        var csv = input.split('-').join(',')
-                       .split(' ').join(',')
-                       .split(':').join(',');
-        var dateArray = csv.split(',');
+      if(typeof input === 'undefined') {
+        return '';
+      }
 
-        var now = new Date();
-        var offset = Math.floor(now.getTimezoneOffset() / 60);
-        var offsetString = (offset < 10 ? '0' + offset : offset.toString()) + ':00';
-        if(offset < 0) {
-            offsetString = '+' + offsetString;
-        }
-        else {
-            offsetString = '-' + offsetString;
-        }
+      var csv = input.split('-').join(',')
+                     .split(' ').join(',')
+                     .split(':').join(',');
+      var dateArray = csv.split(',');
 
-        return dateArray[2] + '-' + dateArray[0] + '-' + dateArray[1] + 'T' + dateArray[3] + ':' + dateArray[4] + offsetString;
+      var now = new Date();
+      var offset = Math.floor(now.getTimezoneOffset() / 60);
+      var offsetString = (offset < 10 ? '0' + offset : offset.toString()) + ':00';
+      if(offset < 0) {
+          offsetString = '+' + offsetString;
+      }
+      else {
+          offsetString = '-' + offsetString;
+      }
+
+      return dateArray[2] + '-' + dateArray[0] + '-' + dateArray[1] + 'T' + dateArray[3] + ':' + dateArray[4] + offsetString;
     }
```

Comment:
I say no, because there is only one format we're parsing here. I made a fix to master yesterday that I have to merge in here.